internal/pkg/print/markdown/content/testdata/content-WithAggregateTypeDefaults.golden

```diff
@@ -0,0 +1,116 @@
+Usage:
+
+module "foo" {
+  source = "github.com/foo/bar"
+
+  id   = "1234567890"
+  name = "baz"
+
+  zones = ["us-east-1", "us-west-1"]
+
+  tags = {
+    Name         = "baz"
+    Created-By   = "first.last@email.com"
+    Date-Created = "20180101"
+  }
+}
+
+## Inputs
+
+These variables are defined:
+
+### unquoted
+
+Description: -
+
+Type: `string`
+
+Default: -
+
+### string-2
+
+Description: It's string number two.
+
+Type: `string`
+
+Default: -
```

Comment:
Same as above. Not sure if in a non-table layout, documenting non-existent default value is of much use. By the way, I suggest you create another variable named `string-3` with `default = ""` to have this use-case covered (which becomes a useful test scenario with this feature).

Reply:
There's one fundamental difference between `document` and the rest of the print types. In `document` when `WithRequired` is selected there will be two different sections:

- `Required Inputs`: doesn't show `Default: -` at all (there's no default value for required inputs)
- `Optional Inputs`: does show `Default: <value>`

But without `--no-required` there will only be one sections, `Inputs`. This is identical to `table` or `json` in terms of showing `Default: -`. I think to make it still consistent within `document` type we can both:

- hide `Default: -` for non-existing `default` Terraform attribute
- show ``Default: `""` `` for _empty_ `default` Terraform attribute

any thought?

Reply:
any thought on this?

Reply:
I like what you suggest. As said before, we haven't captured the use-case of empty strings in `variables.tf` yet, which we should definitely do. A quick test shows... 

1) that a variable that defaults to an empty string results in `` in the table layout
2) compare this with `string-1`, which results in `bar`, not `"bar"`

I'd say that `` isn't as obvious a depiction of an empty string as `""` is. The JSON output of `--with-aggregate-type-defaults` can be regarded as an extension to the default view that only shows concrete default values for variables of type `string`, such as `bar`. If the variable was a `map`, we would show `{ "foo": "bar" }`, so why not show defaults of type strings as `"bar"` instead of `bar` in all markdown documents?

Reply:
Suggestion: to keep this feature focused, I'd produce Default: `` now and later provide a fix that turns `` into `""` for all markdown layouts (please look at the source of this comment to understand what I mean).